lib/bloc/user_profile/user_profile_bloc.dart

```diff
@@ -361,13 +362,31 @@ class UserProfileBloc {
         var userID = token;
         userToRegister = userToRegister.copyWith(token: token, userID: userID);
       }
+      userToRegister = await _restoreUserPreferences(userToRegister);
     } catch (e) {
       _registrationController.addError(e);
     }
     userToRegister = userToRegister.copyWith(registrationRequested: true);
     await _saveChanges(preferences, userToRegister);
   }
 
+  Future<BreezUserModel> _restoreUserPreferences(
```

Comment:
Not sure it is a good idea to leak backup implementation details to here. I think a better approach would be to read the user preferences in the backup bloc.